src/main/kotlin/tw/waterballsa/alpha/wsabot/Main.kt

```diff
@@ -1,27 +1,50 @@
 package tw.waterballsa.alpha.wsabot
 
-import ch.qos.logback.core.util.OptionHelper.getEnv
+import dev.kord.common.entity.Permission
+import dev.kord.common.entity.Permissions
 import dev.kord.core.event.message.MessageCreateEvent
-import dev.kord.gateway.Intent
-import dev.kord.gateway.Intents
+import dev.kord.gateway.*
 import me.jakejmattson.discordkt.dsl.bot
 import me.jakejmattson.discordkt.dsl.listeners
-import tw.waterballsa.alpha.wsabot.knowledgeking.commands.KingofQuizCommand
+import java.util.*
 
 // 1. declare commands
 // 2. listen to events
 // 3. control bot
+@OptIn(PrivilegedIntent::class)
 fun main(args: Array<String>) {
-    bot(getEnv("BOT_TOKEN")) {
-        prefix { "/" } // ?
+    bot(getBotToken()) {
+        prefix { "+" } // ?
         configure {
-            intents = Intents(Intent.GuildMessageReactions, Intent.DirectMessagesReactions)
+            intents = Intents(
+                Intent.GuildMessageReactions,
+                Intent.DirectMessagesReactions,
+                Intent.GuildScheduledEvents,
+                Intent.GuildVoiceStates,
+                Intent.GuildMembers,
+                Intent.Guilds,
+            )
+            defaultPermissions = Permissions(
+                Permission.All,
+                Permission.ViewGuildInsights,
+                Permission.ViewChannel,
+                Permission.ManageChannels
+            )
         }
+
     }
 }
 
+fun getBotToken(): String {
+    val properties = Properties()
+    val file = Thread.currentThread().contextClassLoader.getResourceAsStream("secret.properties")
+    properties.load(file)
+    return properties.getProperty("token")
+}
+
 fun logListeners() = listeners {
     on<MessageCreateEvent> {
-        println(message.content)
+        if (!(message.author?.isBot!!))
+            println(message.content)
     }
-}
+}
```

Comment:
No new line at the end of the file. 
<img width="924" alt="image" src="https://user-images.githubusercontent.com/88727350/215262043-2d15ce26-9710-4213-99fd-8607ccf600e9.png">